usr/src/uts/common/fs/zfs/zil.c

```diff
@@ -908,9 +918,13 @@ zil_lwb_write_init(zilog_t *zilog, lwb_t *lwb)
 		    ZIO_FLAG_CANFAIL);
 	}
 	if (lwb->lwb_zio == NULL) {
+		if (zilog->zl_cur_used <= zil_slog_limit || !lwb->lwb_slog)
+			prio = ZIO_PRIORITY_SYNC_WRITE;
+		else
+			prio = ZIO_PRIORITY_ASYNC_WRITE;
```

Comment:
> Perhaps a more general solution is to add a separate slog-based priority?
Perhaps. My original idea indeed was just to run them with lower priority, but since we have no such mechanism now I've implemented it this way.  I don't mind it to be changed later, it just seems too invasive to do in one patch.

> so taking the async priority to get past the max limit of sync priority seems marginal at best.
The idea was no to get past the limit, but exactly opposite to limit large write burst more strictly to not let it interfere with other really time-critical load.

>Also, can you explain how this is expected to work with large recordsize?
What exactly is your question?  ZIL code has nothing to do with recordsize (aside activation threshold for indirect write, but that is mostly about small recordsize not large). It operates in context of user write operations that can be of any size and alignment.  So even before this patch ZIL code was able to split large write operations between several log records.  The new code just improves space efficiency of this, making for large transactions all log blocks full, except possibly the last one.